general_messaging_constructs.md

```diff
@@ -4,9 +4,75 @@
 * Threading
 * Message types / families
 * Reliable Delivery
-* Attachments
 * Attachment/Payload Signatures
 * Timing
 * ACKs
 * Errors
 * Tracing / troubleshooting
+
+## Attachments
+
+### Summary
+
+DIDComm messages use a structured format with a defined schema and a small inventory of scalar data types (string, number, date, etc). However, it will be quite common for messages to supplement formalized exchange with arbitrary data--images, documents, or types of media not yet invented.
+
+We need a way to "attach" such content to DIDComm messages. This method must be flexible, powerful, and usable without requiring new schema updates for every dynamic variation.
+
+### Reference
+
+Attachments are contained within a list in the `attachments` header.
+
+Each attachment is contained within the following structure.
+
+- `id`: Identifies attached content within the scope of a given message. Recommended on appended attachment descriptors. Possible but generally unused on embedded attachment descriptors. Never required if no references to the attachment exist; if omitted, then there is no way to refer to the attachment later in the thread, in error messages, and so forth. Because `id` is used to compose URIs, it is recommended that this name be brief and avoid spaces and other characters that require URI escaping.
+- `description`: An optional human-readable description of the content.
+- `filename`: A hint about the name that might be used if this attachment is persisted as a file. It is not required, and need not be unique. If this field is present and `mime-type` is not, the extension on the filename may be used to infer a MIME type.
+- `mime_type`: Describes the MIME type of the attached content. Optional but recommended.
```

Comment:
I suggest we go with "media_type" instead of "mime_type", since this is now the preferred verbiage in IETF RFC-land (see https://tools.ietf.org/html/rfc2046).

Reply:
+1 Additionally, let's specify it in a way that references RFC2046